File: 1295.find-numbers-with-even-number-of-digits.java
/*
 * @lc app=leetcode id=1295 lang=java
 *
 * [1295] Find Numbers with Even Number of Digits
 */

// @lc code=start
class Solution {
    private int digits(int x) {
        int ret = 0;
        while (x != 0) {
            ret += 1;
            x /= 10;
        }
        return ret;
    }

    public int findNumbers(int[] nums) {
        int cnt = 0;
        for (int ix = 0; ix != nums.length; ++ix) {
            if (digits(nums[ix]) % 2 == 0)
                cnt += 1;
        }
        return cnt;
    }
}
// @lc code=end

File: 1295.find-numbers-with-even-number-of-digits.cpp
/*
 * @lc app=leetcode id=1295 lang=cpp
 *
 * [1295] Find Numbers with Even Number of Digits
 */

// @lc code=start
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int findNumbers(vector<int>& nums) {
        auto check = [](int n) -> bool {
            return (n >= 10 && n <= 99)
                || (n >= 1000 && n <= 9999)
                || (n >= 100000 && n <= 999999);
        };
        return count_if(nums.begin(), nums.end(), check);
    }
};
// @lc code=end
